custom_components/ha-glocaltokens/api.py

```diff
@@ -54,37 +53,42 @@ def get_android_id(self):
         return self._client.get_android_id()
 
     def create_url(self, ip, port, api_endpoint):
+        """Creates url to endpoint.
+        Note: port argument is unused because all request must be done to 8443"""
         url = "https://{ip}:{port}/{endpoint}".format(
             ip=ip, port=str(PORT), endpoint=api_endpoint
         )
         return url
 
     def get_alarms_and_timers_from(self, device, endpoint):
+        """Fetches timers and alarms from google device"""
         url = self.create_url(device.ip, device.port, endpoint)
         _LOGGER.debug(
             "For device {device} - {url}".format(device=device.device_name, url=url)
         )
+        # verify=False is need to avoid SSL security checks. Othervise it will fail to connect"""
         HEADERS[HEADER_CAST_LOCAL_AUTH] = device.local_auth_token
-        response = requests.get(
-            url, headers=HEADERS, verify=False, timeout=TIMEOUT
-        )  # verify=False is need to avoid SSL security checks. Othervise it will fail to connect
+        response = requests.get(url, headers=HEADERS, verify=False, timeout=TIMEOUT)
 
         if response.status_code != HTTP_OK:
             _LOGGER.error(
-                "For device {device} - API returned {error}".format(
-                    device=device.device_name, error=response.status_code
+                "For device {device} - API returned {error}: {reason}".format(
+                    device=device.device_name,
+                    error=response.status_code,
+                    reason=response.reason,
```

Comment:
Are you sure the error message lands in `response.reason`? Some other http requests i have worked with were returning error message in `response.message`, but i m not sure what would be here.

Reply:
As far as i can see requests lib does not return message but instead reason if it fails [requests-lib](https://requests.readthedocs.io/en/master/api/#requests.Response)

We can also change it to `response.text` instead?

Reply:
I have changed it to `response.text` because it will return the most amount of info in case of error